agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -0,0 +1,617 @@
+			seelog.Debugf("Skipping cleanup of local credentialspec file option: %s", key)
+			continue
+		}
+		credSpecSplit := strings.SplitAfterN(value, "credentialspec=", 2)
+		localCredentialSpecFile := credSpecSplit[1]
+		err := cs.os.Remove(localCredentialSpecFile)
+		if err != nil {
+			seelog.Debugf("Unable to clear local credential spec file %s for task %s", localCredentialSpecFile, cs.taskARN)
+		}
+		delete(cs.CredSpecMap, key)
+	}
+}
+
+// CredentialSpecResourceJSON is the json representation of the credentialspec resource
+type CredentialSpecResourceJSON struct {
+	TaskARN                 string                               `json:"taskARN"`
+	CreatedAt               *time.Time                           `json:"createdAt,omitempty"`
+	DesiredStatus           *CredentialSpecStatus                `json:"desiredStatus"`
+	KnownStatus             *CredentialSpecStatus                `json:"knownStatus"`
+	RequiredCredentialSpecs map[string][]*apicontainer.Container `json:"credentialSpecResources"`
+	CredSpecMap             map[string]string                    `json:"CredSpecMap"`
+	ExecutionCredentialsID  string                               `json:"executionCredentialsID"`
+}
+
+// MarshalJSON serialises the CredentialSpecResourceJSON struct to JSON
+func (cs *CredentialSpecResource) MarshalJSON() ([]byte, error) {
+	if cs == nil {
+		return nil, errors.New("credential specresource is nil")
+	}
+	createdAt := cs.GetCreatedAt()
+	return json.Marshal(CredentialSpecResourceJSON{
+		TaskARN:   cs.taskARN,
+		CreatedAt: &createdAt,
+		DesiredStatus: func() *CredentialSpecStatus {
+			desiredState := cs.GetDesiredStatus()
+			s := CredentialSpecStatus(desiredState)
+			return &s
+		}(),
+		KnownStatus: func() *CredentialSpecStatus {
+			knownState := cs.GetKnownStatus()
+			s := CredentialSpecStatus(knownState)
+			return &s
+		}(),
+		RequiredCredentialSpecs: cs.getRequiredCredentialSpecs(),
+		CredSpecMap:             cs.getCredSpecMap(),
+		ExecutionCredentialsID:  cs.getExecutionCredentialsID(),
+	})
+}
+
+// UnmarshalJSON deserialises the raw JSON to a CredentialSpecResourceJSON struct
+func (cs *CredentialSpecResource) UnmarshalJSON(b []byte) error {
+	temp := CredentialSpecResourceJSON{}
+
+	if err := json.Unmarshal(b, &temp); err != nil {
+		return err
+	}
+
+	if temp.DesiredStatus != nil {
+		cs.SetDesiredStatus(resourcestatus.ResourceStatus(*temp.DesiredStatus))
+	}
+	if temp.KnownStatus != nil {
+		cs.SetKnownStatus(resourcestatus.ResourceStatus(*temp.KnownStatus))
+	}
+	if temp.CreatedAt != nil && !temp.CreatedAt.IsZero() {
+		cs.SetCreatedAt(*temp.CreatedAt)
+	}
+	if temp.RequiredCredentialSpecs != nil {
+		cs.requiredCredentialSpecs = temp.RequiredCredentialSpecs
+	}
+	cs.taskARN = temp.TaskARN
+	cs.executionCredentialsID = temp.ExecutionCredentialsID
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) setCredentialSpecResourceLocation() error {
+	// TODO: Use registry to setup credentialspec resource location
```

Comment:
Do you have anywhere to track this todo? And what is the plan for you guys to switch to registry? Otherwise, this TODO in our code base may end up nowhere.

Reply:
Yes, we do. This is on the list of items that will be a fast follow up.